frontend/packages/ceph-storage-plugin/src/components/volume-snapshot/volume-snapshot.tsx

```diff
@@ -0,0 +1,233 @@
+
+const snapshotMenuActions = [...getKebabActionsForKind(VolumeSnapshotModel)];
+
+const snapshotTableColumnClasses = [
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm', 'pf-u-w-16-on-lg'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-lg'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-xl'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-2xl'),
+  Kebab.columnClass,
+];
+
+const VolumeSnapshotTableHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[0] },
+    },
+    {
+      title: 'Date',
+      sortField: 'metadata.creationTimestamp',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[1] },
+    },
+    {
+      title: 'Status',
+      sortField: 'status.phase',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[2] },
+    },
+    {
+      title: 'Size',
+      sortField: 'status.capacity.storage',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[3] },
+    },
+    {
+      title: 'Labels',
+      sortField: 'metadata.labels',
+      transforms: [sortable],
+      props: { className: snapshotTableColumnClasses[4] },
+    },
+    {
+      title: '',
+      props: { className: snapshotTableColumnClasses[5] },
+    },
+  ];
+};
+VolumeSnapshotTableHeader.displayName = 'SnapshotTableHeader';
+
+const kind = referenceFor(VolumeSnapshotModel);
+
+export const VolumeSnapshotDetails: React.SFC = (props) => {
+  const [resource, setResource] = React.useState(null);
+
+  React.useEffect(() => {
+    k8sGet(VolumeSnapshotModel, _.get(props, 'match.params.name'), _.get(props, 'match.params.ns'))
+      .then((volumeSnapshotObj) => {
+        try {
+          setResource(volumeSnapshotObj);
+        } catch (e) {
+          setResource(null);
+        }
+      })
+      .catch((error) => {
+        setResource(null);
+        throw error;
+      });
+  }, [props]);
+
+  return (
+    <div>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          <BreadCrumbs
+            breadcrumbs={[
+              {
+                name: PersistentVolumeClaimModel.labelPlural,
+                path: `/k8s/ns/${_.get(props, 'match.params.ns')}/persistentvolumeclaims`,
+              },
+              {
+                name: `${_.get(props, 'match.params.appName')}`,
+                path: `/k8s/ns/${_.get(props, 'match.params.ns')}/persistentvolumeclaims/${_.get(
+                  props,
+                  'match.params.appName',
+                )}`,
+              },
+              {
+                name: 'Snapshot Details',
+                path: ``,
+              },
+            ]}
+          />
+        </div>
+
+        <h2 className="co-m-pane__heading co-m-pane__heading--baseline">
```

Comment:
You shouldn't need all this markup. You should be able to reuse the shared components for other details pages.